File: 167. Two Sum II - Input Array Is Sorted.go
func twoSum(numbers []int, target int) []int {
    left, right := 0, len(numbers)-1
    var res []int
    fmt.Println(numbers[left], numbers[right])
    for left < right{
        if (numbers[left] + numbers[right] > target) {
            right -= 1
        }else if (numbers[left] + numbers[right] < target){
            left += 1
        } else{
            res = append(res, left+1)
            res = append(res, right+1)
            return res
        }
    }
    return []int{}    
}

File: 167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        right, left = 0, len(numbers)-1
        def calSum(right, left):
            addition = numbers[right] + numbers[left]
            if addition > target:
                left -= 1
            elif addition < target:
                right += 1
            else:
                return [right+1, left+1]
            return calSum(right, left)
        return calSum(right, left)
        